client/packages/system/src/Immunisation/VaccineCourseView/VaccineCourseView.tsx

```diff
@@ -121,127 +99,119 @@ const Row = ({
   </Box>
 );
 
-const createNewVaccineCourse = (seed?: any | null): any => ({
-  id: FnUtils.generateUUID(),
-  name: '',
-  description: '',
-  ...seed,
-});
-
-interface UseDraftVaccineCourseControl {
-  draft: any;
-  onUpdate: (patch: Partial<any>) => void;
-  onSave: () => Promise<void>;
-  isLoading: boolean;
-}
-
-// dummy vaccine items
-const VaccineOptions = [
+// TODO replace with item querr
+const VaccineOptions: VaccineCourseItemNode[] = [
   {
-    label: 'vaccine 1',
-    value: 'vaccine 1',
+    id: 'vaccine 1',
+    itemId: 'vaccine 1',
+    __typename: 'VaccineCourseItemNode',
   },
   {
-    label: 'vaccine 2',
-    value: 'vaccine 2',
+    id: 'vaccine 2',
+    itemId: 'vaccine 2',
+    __typename: 'VaccineCourseItemNode',
   },
 ];
 
-const useDraftVaccineCourse = (): UseDraftVaccineCourseControl => {
-  const [vaccine, setVaccineCourse] = useState<any>(() =>
-    createNewVaccineCourse(seed)
-  );
+export const VaccineCourseView: FC = () => {
+  const { setSuffix, navigateUpOne } = useBreadcrumbs();
+  const t = useTranslation('coldchain');
+  const { id } = useParams();
+  const {
+    draft,
+    update: { update },
+    updatePatch,
+    query: { data, isLoading },
+    isDirty,
+  } = useVaccineCourse(id);
+  const { data: demographicData } = useDemographicIndicators();
 
-  const onUpdate = (patch: Partial<any>) => {
-    setVaccineCourse({ ...vaccine, ...patch });
-  };
+  const [buffer, setBuffer] = useState(draft?.doses ?? 1);
+  const [value, setValue] = useState(draft?.doses ?? 1);
```

Comment:
Can you add a comment about what this is needed?
Maybe a better name too. value's pretty generic.
Maybe `doseCount` and `bufferedDoseCount`?